sdk/core/loader/boot.cc

```diff
@@ -1004,6 +933,17 @@ extern "C" SchedulerEntryInfo loader_entry_point(const ImgHdr &imgHdr,
                                                  void         *almightySeal,
                                                  void         *almightyRW)
 {
+	// This relies on a slightly surprising combination of two C++ features:
```

Comment:
Can you expand on what "This" is?  I'm sure it's quite informative, but this comment (or, at least, the `SchedulerEntryInfo` type) seems unrelated to the rest of the PR?